content/courses/intro-to-spark/chapters/05_Proof_Of_Functional_Correctness.rst

```diff
@@ -542,6 +542,11 @@ the proof or strip it down to a small number of easily-reviewable
 assumptions. For this purpose, you can add assertions to break complex
 proofs into smaller steps.
 
+.. todo::
+    
+    This might confuse people, what is the difference between the
```

Comment:
The difference is that the first assertion is proved, the second is not, see the comments in the code.

Reply:
What confuses me here is what ```Assertion_Checked_By_The_Tool``` and what ```Assumption_Validated_By_Other_Means``` are. 

I think these are placeholders for some sort of boolean assertion expressions?

Reply:
yes, the above are just placeholders for arbitrary boolean expressions :-)

Reply:
I do think examples could be useful here.

What I didn't really understand after reading this section was what kinds of expressions can be used for ```Assertion_Checked_By_The_Tool``` and ```Assumption_Validated_By_Other_Means```. 

I definitely acknowledge this is a complex topic which I may not have devoted enough time to, thus the reader could be the issue here as well :-P